### PR TITLE
[Perpetual]: Add new message to add collateral to existing position

### PR DESCRIPTION
**Pull request by amityadav0**
## Description

- In current implementation there is no way to add collateral to existing position with leverage = 1, This is needed for user so that they can improve the position health

### What has Changed?

- Add a new message to add collateral to existing position to improve position health

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/elys-network/elys/blob/main/docs/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#testing)
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### Deployment Notes

Are there any specific considerations to take into account when deploying these changes? This may include new dependencies, scripts that need to be executed, or any aspects that can only be evaluated in a deployed environment.

### Screenshots and Videos

Please provide any relevant before and after screenshots by uploading them here. Additionally, demo videos can be highly beneficial in demonstrating the process.

### CODE REVIEW

**Comment by amityadav0** [comment]
I have followed this approach:

When a user adds a collateral, liability will decrease, custody remains same, collateral will increase

File: x/perpetual/keeper/msg_server_add_collateral.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/elys-network/elys/x/perpetual/types"
+)
+
+func (k msgServer) AddCollateral(goCtx context.Context, msg *types.MsgAddCollateral) (*types.MsgAddCollateralResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// TODO: Handling the message
+	_ = ctx
```
Comment (cosmic-vagabond) [inline x/perpetual/keeper/msg_server_add_collateral.go:14]: @amityadav0 that one does not seem to implement any functionality

File: x/perpetual/keeper/msg_server_broker_add_collateral.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	}
+
+	// fail if msg.Creator is not broker address
+	if msg.Creator != k.parameterKeeper.GetParams(ctx).BrokerAddress {
+		return nil, errors.Wrap(types.ErrUnauthorised, "creator must be broker address")
+	}
+
+	// TODO
+	//msgAddCollateral := types.NewMsgAddCollateral(msg.Owner, msg.Amount, uint64(msg.Id))
+	//return k.Keeper.Add(ctx, msgAddCollateral)
```
Comment (cosmic-vagabond) [inline x/perpetual/keeper/msg_server_broker_add_collateral.go:26]: @amityadav0 that one does not seem to implement any functionality

File: x/perpetual/keeper/msg_server_add_collateral.go (hunk 1 above)
Comment (amityadav0) [reply x/perpetual/keeper/msg_server_add_collateral.go]: updated

File: x/perpetual/keeper/msg_server_broker_add_collateral.go (hunk 2 above)
Comment (amityadav0) [reply x/perpetual/keeper/msg_server_broker_add_collateral.go]: updated

**Review by cosmic-vagabond — APPROVED** [review]
LGTM

**Pull request merged**